ouroboros-consensus/test/consensus-test/Test/Consensus/Util/MonadSTM/NormalForm.hs

```diff
@@ -0,0 +1,58 @@
+{-# LANGUAGE DeriveAnyClass     #-}
+{-# LANGUAGE DeriveGeneric      #-}
+{-# LANGUAGE DerivingStrategies #-}
+{-# LANGUAGE RankNTypes         #-}
+{-# LANGUAGE TypeApplications   #-}
+module Test.Consensus.Util.MonadSTM.NormalForm (tests) where
+
+import           Control.Monad.IOSim
+import           GHC.Generics
+import           NoThunks.Class
+import           Ouroboros.Consensus.Util.MonadSTM.NormalForm (MonadSTM,
+                     newMVar, updateMVar)
+import           Test.Tasty
+import           Test.Tasty.QuickCheck
+
+
+-- Note that all of the tests here are only significant with compiler
+-- optimizations turned off!
```

Comment:
I understand what you mean to say here, but maybe elaborate on why we are adding these tests. This will make it clear for future readers why running the tests without compiler optimisations turned off could be a problem

Reply:
I've added a larger description of what the tests do and why they aren't useful with compiler optimizations turned on 👍